### ISSUE TITLE
findJobMetrics.py raised HTTP_EXCEPTIONS using cast-log-ras-* index

### ISSUE THREAD

**Issue by thanh-lam** [IBM/CAST#378]
**Describe the bug**
findJobMetrics.py failed with following error when using cast-log-ras-* index. With other indices such as cast-log-syslog-*, it just produced empty list with no error.
```
[root@bd01 ~]# findJobMetrics.py -j 540 -f "msg_id" -i "cast-log-ras-*"         Got 1 Hit(s) for specified job:
Traceback (most recent call last):
  File "/opt/ibm/csm/bigdata/python/findJobMetrics.py", line 210, in <module>
    sys.exit(main(sys.argv))
  File "/opt/ibm/csm/bigdata/python/findJobMetrics.py", line 157, in main
    body=body
  File "/usr/lib/python2.7/site-packages/elasticsearch/client/utils.py", line 76, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/usr/lib/python2.7/site-packages/elasticsearch/client/__init__.py", line 660, in search
    doc_type, '_search'), params=params, body=body)
  File "/usr/lib/python2.7/site-packages/elasticsearch/transport.py", line 318, in perform_request
    status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
  File "/usr/lib/python2.7/site-packages/elasticsearch/connection/http_urllib3.py", line 186, in perform_request
    self._raise_error(response.status, raw_data)
  File "/usr/lib/python2.7/site-packages/elasticsearch/connection/base.py", line 125, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
elasticsearch.exceptions.RequestError: RequestError(400, u'search_phase_execution_exception', u'Fielddata is disabled on text fields by default. Set fielddata=true on [msg_id] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memory. Alternatively use a keyword field instead.')
```

**To Reproduce**
1. Go to `/opt/ibm/csm/bigdata/python/`
2. Run `./findJobMetrics.py -j 540 -f "msg_id" -i "cast-log-ras-*"`
3. See error

**Expected behavior**
Run `./findJobMetrics.py -j 540 -f "msg_id" -i "cast-log-syslog-*"` and see following output. Note that the list is empty. It should show data about the msg_id field which can be seen in kibana.
```
[root@bd01 ~]# findJobMetrics.py -j 540 -f "msg_id" -i "cast-log-syslog-*"
Got 1 Hit(s) for specified job:

Metric Analysis for Job ID 540 - 0 :

     Field |            Min |            Max |        Average |        Std Dev | Count
```

**Screenshots**

**Environment (please complete the following information):**
 - Machine bd01 (IST)
 - Version CSM 1.3.0 1401

**Additional context**

**Issue Source:**
Testing the BDS Use Cases in IST.

**Comment by mew2057** [comment]
This is not so much a problem with the usecase, but the index itself. The best we'll be able to do is clean up the error message.

**Comment by thanh-lam** [comment]
These are actually elasticsearch errors. Found similar messages in elasticsearch.log. Can we set `fielddata=true` as suggested?
```
org.elasticsearch.transport.RemoteTransportException: [xxxbd01][x.x.x.x:9300][indices:data/read/search[phase/query]]
Caused by: java.lang.IllegalArgumentException: Fielddata is disabled on text fields by default. Set fielddata=true on [msg_id] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memory. Alternatively use a keyword field instead.
        at org.elasticsearch.index.mapper.TextFieldMapper$TextFieldType.fielddataBuilder(TextFieldMapper.java:518) ~[elasticsearch-6.3.2.jar:6.3.2]
        at org.elasticsearch.index.fielddata.IndexFieldDataService.getForField(IndexFieldDataService.java:115) ~[elasticsearch-6.3.2.jar:6.3.2]
        at org.elasticsearch.index.query.QueryShardContext.getForField(QueryShardContext.java:165) ~[elasticsearch-6.3.2.jar:6.3.2]
        at org.elasticsearch.search.DefaultSearchContext.getForField(DefaultSearchContext.java:501) ~[elasticsearch-6.3.2.jar:6.3.2]
        at org.elasticsearch.search.aggregations.support.MultiValuesSourceAggregationBuilder.config(MultiValuesSourceAggregationBuilder.java:293) ~[?:?]
        at org.elasticsearch.search.aggregations.support.MultiValuesSourceAggregationBuilder.resolveConfig(MultiValuesSourceAggregationBuilder.java:252) ~[?:?]
        at org.elasticsearch.search.aggregations.support.MultiValuesSourceAggregationBuilder.doBuild(MultiValuesSourceAggregationBuilder.java:244) ~[?:?]
        at org.elasticsearch.search.aggregations.support.MultiValuesSourceAggregationBuilder$LeafOnly.doBuild(MultiValuesSourceAggregationBuilder.java:52) ~[?:?]
...
```

**Comment by mew2057** [comment]
I made a fix to improve error reporting. As for the specific problem I think using the generated keyword field might be the better option.

**Comment by thanh-lam** [comment]
Verified the command with latest build, 1.4.0-1690. The HTTP exception errors are gone. But, the message just left me wonder -- It got 3 hits but it only supports 1 hit (?)
```
[root@bd01 python]# ./findJobMetrics.py -j 125 -f "msg_id" -i "cast-log-ras-*"
Got 3 Hit(s) for specified job:
This implementation only supports queries where the hit count is equal to 1.
```
The command `findJobMetrics.py` is pending on another issue to be fixed anyway.

**Comment by mew2057** [comment]
`findJobMetrics.py` doesn't support ambiguous queries, in this case this means there's more than one job with the job id of 125.

What other issue is it pending on?

**Comment by thanh-lam** [comment]
#340 Difference in elasticsearch @timestamp and times logged in data.

**Comment by thanh-lam** [comment]
Verified with CSM 1.4.0-1756. Those data fields are now set to 0. Closing.
```
"data": {
    "gpfs_ns_tot_queue_wait_rd": 0,
    "gpfs_ns_tot_queue_wait_wr": 0,
    "gpfs_ns_bytes_read":0,
    ...
    "gpfs_ns_byte_written": 0
}
```
